Asterix/wfsc/wf_control_functions.py

```diff
@@ -488,11 +492,14 @@ def create_singlewl_interaction_matrix(testbed: Testbed,
                     else:
 
                         wavefront = crop_or_pad_image(
-                            prop.prop_angular_spectrum(
-                                wavefrontupstreaminDM * DM.EF_from_phase_and_ampl(
-                                    phase_abb=phase_in_basis + DM_phase_init[testbed.name_of_DMs.index(DM_name)],
-                                    wavelengths=wavelength), wavelength, -DM.z_position, DM.diam_pup_in_m / 2, DM.prad),
-                            DM.dim_overpad_pupil)
+                            prop.prop_angular_spectrum(wavefrontupstreaminDM * DM.EF_from_phase_and_ampl(
```

Comment:
Could you assign the return from `DM.EF_from_phase_and_ampl` to a variable and feed that into `prop_angular_spectrum()`? The way it is written right now is incredibly hard to understand in terms of which inputs go into what function.

Reply:
done